PR: https://github.com/odoo/odoo/pull/178200

From: f10bd8ddcf99c9247dca417fefbc260454d74ea1
From: Gorash
Date: 2024-10-15 09:33:02

Breaking data model changes scores: del:8 + add:0, change matches:
-    _inherit = 'barcode.rule'
-    _inherit = 'loyalty.mail'
-    _inherit = 'pos.config'
-    _inherit = 'pos.order'
-    _inherit = 'pos.order.line'
-    _inherit = 'pos.session'
-    _inherit = 'product.product'
-    _inherit = 'res.partner'

Total Changes: 29

[IMP] *: adapt model class names to correspond to model names (apply script)

Part-of: odoo/odoo#178200
Related: odoo/enterprise#69762
Related: odoo/documentation#11115
Related: odoo/upgrade#6577
Related: odoo/design-themes#988
Related: odoo/upgrade-util#143

================================= pseudo patch: =================================

--- a/addons/pos_loyalty/models/barcode_rule.py
+++ b/addons/pos_loyalty/models/barcode_rule.py
@@ -5,6 +5,6 @@ from odoo import models, fields
 
 
 class BarcodeRule(models.Model):
-    _inherit = 'barcode.rule'
+    _inherit = ['barcode.rule']
 
     type = fields.Selection(selection_add=[('coupon', 'Coupon')], ondelete={'coupon': 'set default'})

--- a/addons/pos_loyalty/models/loyalty_card.py
+++ b/addons/pos_loyalty/models/loyalty_card.py
@@ -3,8 +3,8 @@
 
 from odoo import fields, models, api
 
+
 class LoyaltyCard(models.Model):
-    _name = 'loyalty.card'
     _inherit = ['loyalty.card', 'pos.load.mixin']
 
     source_pos_order_id = fields.Many2one('pos.order', "PoS Order Reference",

--- a/addons/pos_loyalty/models/loyalty_mail.py
+++ b/addons/pos_loyalty/models/loyalty_mail.py
@@ -3,8 +3,9 @@
 
 from odoo import fields, models
 
+
 class LoyaltyMail(models.Model):
-    _inherit = 'loyalty.mail'
+    _inherit = ['loyalty.mail']
 
     pos_report_print_id = fields.Many2one('ir.actions.report', string="Print Report", domain=[('model', '=', 'loyalty.card')],
         help="The report action to be executed when creating a coupon/gift card/loyalty card in the PoS.",

--- a/addons/pos_loyalty/models/loyalty_program.py
+++ b/addons/pos_loyalty/models/loyalty_program.py
@@ -3,8 +3,8 @@
 from odoo import _, api, fields, models
 from odoo.exceptions import UserError
 
+
 class LoyaltyProgram(models.Model):
-    _name = 'loyalty.program'
     _inherit = ['loyalty.program', 'pos.load.mixin']
 
     # NOTE: `pos_config_ids` satisfies an excpeptional use case: when no PoS is specified, the loyalty program is

--- a/addons/pos_loyalty/models/loyalty_reward.py
+++ b/addons/pos_loyalty/models/loyalty_reward.py
@@ -5,8 +5,8 @@ from odoo import models, api
 import ast
 import json
 
+
 class LoyaltyReward(models.Model):
-    _name = 'loyalty.reward'
     _inherit = ['loyalty.reward', 'pos.load.mixin']
 
     def _get_discount_product_values(self):

--- a/addons/pos_loyalty/models/loyalty_rule.py
+++ b/addons/pos_loyalty/models/loyalty_rule.py
@@ -4,8 +4,8 @@
 from odoo import api, fields, models
 from odoo.osv import expression
 
+
 class LoyaltyRule(models.Model):
-    _name = 'loyalty.rule'
     _inherit = ['loyalty.rule', 'pos.load.mixin']
 
     valid_product_ids = fields.Many2many(

--- a/addons/pos_loyalty/models/pos_config.py
+++ b/addons/pos_loyalty/models/pos_config.py
@@ -4,8 +4,9 @@
 from odoo import _, fields, models
 from odoo.exceptions import UserError
 
+
 class PosConfig(models.Model):
-    _inherit = 'pos.config'
+    _inherit = ['pos.config']
 
     # NOTE: this funtions acts as a m2m field with loyalty.program model. We do this to handle an excpetional use case:
     # When no PoS is specified at a loyalty program form, this program is applied to every PoS (instead of none)

--- a/addons/pos_loyalty/models/pos_order.py
+++ b/addons/pos_loyalty/models/pos_order.py
@@ -6,8 +6,9 @@ from odoo import _, models
 from odoo.tools import float_compare
 import base64
 
+
 class PosOrder(models.Model):
-    _inherit = 'pos.order'
+    _inherit = ['pos.order']
 
     def validate_coupon_programs(self, point_changes, new_codes):
         """

--- a/addons/pos_loyalty/models/pos_order_line.py
+++ b/addons/pos_loyalty/models/pos_order_line.py
@@ -3,8 +3,9 @@
 
 from odoo import fields, models, api
 
+
 class PosOrderLine(models.Model):
-    _inherit = 'pos.order.line'
+    _inherit = ['pos.order.line']
 
     is_reward_line = fields.Boolean(
         help="Whether this line is part of a reward or not.")

--- a/addons/pos_loyalty/models/pos_session.py
+++ b/addons/pos_loyalty/models/pos_session.py
@@ -2,8 +2,9 @@
 
 from odoo import models, api
 
+
 class PosSession(models.Model):
-    _inherit = 'pos.session'
+    _inherit = ['pos.session']
 
     @api.model
     def _load_pos_data_models(self, config_id):

--- a/addons/pos_loyalty/models/product_product.py
+++ b/addons/pos_loyalty/models/product_product.py
@@ -2,7 +2,7 @@ from odoo import api, models
 
 
 class ProductProduct(models.Model):
-    _inherit = 'product.product'
+    _inherit = ['product.product']
 
     @api.model
     def _load_pos_data_fields(self, config_id):

--- a/addons/pos_loyalty/models/res_partner.py
+++ b/addons/pos_loyalty/models/res_partner.py
@@ -2,6 +2,6 @@ from odoo import fields, models
 
 
 class ResPartner(models.Model):
-    _inherit = 'res.partner'
+    _inherit = ['res.partner']
 
     loyalty_card_count = fields.Integer(groups='base.group_user,point_of_sale.group_pos_user')
